cmd/libs/go2idl/client-gen/main.go

```diff
@@ -18,45 +18,93 @@ limitations under the License.
 package main
 
 import (
+	"fmt"
+	"path/filepath"
+
 	"k8s.io/kubernetes/cmd/libs/go2idl/args"
 	"k8s.io/kubernetes/cmd/libs/go2idl/client-gen/generators"
+	"k8s.io/kubernetes/pkg/api/unversioned"
 
 	"github.com/golang/glog"
 	flag "github.com/spf13/pflag"
 )
 
-var test = flag.BoolP("test", "t", false, "set this flag to generate the client code for the testdata")
+var (
+	test          = flag.BoolP("test", "t", false, "set this flag to generate the client code for the testdata")
+	inputVersions = flag.StringSlice("input", []string{"api/", "extensions/"}, "group/versions that client-gen will generate clients for. At most one version per group is allowed. Specified in the format \"group1/version1,group2/version2...\". Default to \"api/,extensions\"")
```

Comment:
We are generating a clientset for the unversioned types right now, because all the user of the client package is using an unversioned one.